CHANGELOG.md

```diff
@@ -1,5 +1,126 @@
+
+This is useful for testing websites that perform authentication via local storage instead of cookies.
+
+#### Miscellaneous
+
+* `StorageSandbox` is shared to the client API with `clear` and `lock` methods added. ([testcafe-hammerhead/#1189](https://github.com/DevExpress/testcafe-hammerhead/issues/1189))
+* The number of failed cookie-sync messages is now limited. ([testcafe-hammerhead/#1193](https://github.com/DevExpress/testcafe-hammerhead/issues/1193))
+* `Hammerhead.backupStorages` and `Hammerhead.restoreStorages` methods are added to the client API([testcafe-hammerhead/#1125](https://github.com/DevExpress/testcafe-hammerhead/issues/1125))
+* The `GetEventListeners` method is implemented in `sandbox.event.listeners`. ([testcafe-hammerhead/#1238](https://github.com/DevExpress/testcafe-hammerhead/issues/1238))
+* JavaScript expressions now supported in raw API selectors. ([#1466](https://github.com/DevExpress/testcafe/issues/1466))
+
+### Bug Fixes
+
+* Authentication with user roles works correctly in SPAs. ([#1502](https://github.com/DevExpress/testcafe/issues/1502))
+* Description for the `path` parameter of the `t.takeScreenshot` action has been corrected ([#1515](https://github.com/DevExpress/testcafe/issues/1515))
+* Local storage is now cleaned appropriately after the test run.([#1546](https://github.com/DevExpress/testcafe/issues/1546))
+* TestCafe now checks element visibility with a timeout when the target element's `style.top` is negative ([#1185](https://github.com/DevExpress/testcafe/issues/1185))
+* Fetching an absolute CORS URL now works correctly. ([#1629](https://github.com/DevExpress/testcafe/issues/1629))
+* Fixed a bug that led to an error in test code. ([#1442](https://github.com/DevExpress/testcafe/issues/1442))
+* TypeScript performance has been enhanced. ([#1591](https://github.com/DevExpress/testcafe/issues/1591))
+* Mousedown on a link without `src` is now handled properly. ([#1598](https://github.com/DevExpress/testcafe/issues/1598))
+* Fixed a bug that led to test failing in Edge 15. ([testcafe-hammerhead/#1176](https://github.com/DevExpress/testcafe-hammerhead/issues/1176))
+* The right port is now applied to a cross-domain iframe location after redirect. ([testcafe-hammerhead/#1191](https://github.com/DevExpress/testcafe-hammerhead/issues/1191))
+* `Hammerhead.EVENTS.xhrSend` and `Hammerhead.EVENTS.fetchSend` are renamed to `...Sent`. ([testcafe-hammerhead/#678](https://github.com/DevExpress/testcafe-hammerhead/issues/678))
+* All internal properties are marked as non-enumerable. ([testcafe-hammerhead/#1182](https://github.com/DevExpress/testcafe-hammerhead/issues/1182))
+* Hammerhead no longer crashes on the `slack.com/signin` page. ([testcafe-hammerhead/#1195](https://github.com/DevExpress/testcafe-hammerhead/issues/1195))
+* Fixed a bug where a script error occurred after proxying. ([testcafe-hammerhead/#1209](https://github.com/DevExpress/testcafe-hammerhead/issues/1209))
```

Comment:
Change to some meaningful title (as I see it was a problem when you stringify something with JSON?) /cc @miherlosev